pybossa/dashboard.py

```diff
@@ -0,0 +1,408 @@
+                       - ('1 week')::INTERVAL GROUP BY day, task_run.user_id)
+                       SELECT user_id, COUNT(user_id) AS n_days
+                       FROM data GROUP BY user_id HAVING(count(user_id) > 1)
+                       ORDER by n_days;
+                          ''')
+            results = db.slave_session.execute(sql)
+            db.session.commit()
+            return "Materialized view created"
+
+def format_users_week():
+    """Return a variable with users data."""
+    session = db.slave_session
+    # Registered users
+    sql = text('''select * from dashboard_week_users''')
+    results = session.execute(sql)
+    labels = []
+    series = []
+    for row in results:
+        labels.append(row.day.strftime('%Y-%m-%d'))
+        series.append(int(row.n_users))
+    if len(labels) == 0:
+        labels.append(datetime.now().strftime('%Y-%m-%d'))
+    if len(series) == 0:
+        series.append(0)
+    active_users_last_week = dict(labels=labels, series=[series])
+    return active_users_last_week
+
+def format_anon_week():
+    """Return a variable with anon data."""
+    session = db.slave_session
+    # Anon users
+    sql = text('''select * from dashboard_week_anon''')
+    results = session.execute(sql)
+    labels = []
+    series = []
+    for row in results:
+        labels.append(row.day.strftime('%Y-%m-%d'))
+        series.append(int(row.n_users))
+    if len(labels) == 0:
+        labels.append(datetime.now().strftime('%Y-%m-%d'))
+    if len(series) == 0:
+        series.append(0)
+    active_anon_last_week = dict(labels=labels, series=[series])
+    return active_anon_last_week
+
+def format_new_projects():
+    """Return new projects data."""
+    session = db.slave_session
+    # New projects
+    sql = text('''select * from dashboard_week_project_new''')
+    results = session.execute(sql)
+    new_projects_last_week = []
+    for row in results:
+        datum = dict(day=row.day, id=row.id, short_name=row.short_name,
+                     p_name=row.name, owner_id=row.owner_id, u_name=row.u_name,
+                     email_addr=row.email_addr)
+        new_projects_last_week.append(datum)
+    return new_projects_last_week
+
+def format_update_projects():
+    """Return updated projects data."""
+    session = db.slave_session
+    # Updated projects
+    sql = text('''select * from dashboard_week_project_update''')
+    results = session.execute(sql)
+    update_projects_last_week = []
+    for row in results:
+        datum = dict(day=row.day, id=row.id, short_name=row.short_name,
+                     p_name=row.name, owner_id=row.owner_id, u_name=row.u_name,
+                     email_addr=row.email_addr)
+        update_projects_last_week.append(datum)
+    return update_projects_last_week
+
+def format_new_tasks():
+    """Return new tasks data."""
+    session = db.slave_session
+    # New task
+    sql = text('''select * from dashboard_week_new_task''')
```

Comment:
That raises an error, which is captured and pass to the template so it shows the data is not available yet ;-)